src/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react';
+import './style.scss';
+import "../../styles/vars.scss";
+import { faTimesCircle } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+class Modal extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state= {
+            open: this.props.open,  // color, type 
+            type: this.props.type,  //entire-screen, in-place
+            background_color: this.props.background_color, //color
+            blur: this.props.blur // true, false
+            
+        }         
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.setState({
+            open: nextProps.open,
+            type: nextProps.type,  
+            background_color: nextProps.background_color,
+            blur: nextProps.blur 
+        })
+    }
+
+    show = () => {
+        document.addEventListener('click', this.handleClick);
+        document.body.classList.add('modal-visible');
+    }
+
+    close = () => {
+        this.setState({ open: false },() => {
+            document.removeEventListener('click', this.handleClick);
+          });
+        document.body.classList.remove('modal-visible');
+        this.props.closeModal();
+    }
+
+
+    handleClick= (e)=> {
+    // if (e.target.className === "modal-wrapper") {
+    //     this.close();
+    // }
+        if (this.node.contains(e.target)){
+            return;
+        }
+
+        this.close();
+    }
+
+    render() {
+
+        if(this.state.open)
+            this.show();
+        const mdl=this.state.type==='entire-screen'?(
```

Comment:
is there a reason why you store this into a const variable instead of returning directly?

Reply:
Nope! It was just trying to make it clearer for me personally! I've changed it to a single return statement now!